map/place_page_info.cpp

```diff
@@ -67,27 +68,58 @@ void Info::SetFromFeatureType(FeatureType & ft)
     m_uiSubtitle = FormatSubtitle(true /* withType */);
     m_uiAddress = m_address;
   }
-  else if (!m_primaryFeatureName.empty())
+  else if (IsPublicTransportStop())
   {
-    m_uiTitle = m_primaryFeatureName;
-    m_uiSecondaryTitle = out.secondary;
-    m_uiSubtitle = FormatSubtitle(true /* withType */);
-    m_uiAddress = m_address;
+    auto const local_ref = GetMetadata(feature::Metadata::FMD_LOCAL_REF);
+
+    if (!local_ref.empty()) // has local ref
+    {
+      std::string localizedFormat = platform::GetLocalizedString("poi_title_local_ref");
+      char titleBuffer [1024];
+
+      if (!m_primaryFeatureName.empty()) // has name and local_ref
+      {
+        snprintf(titleBuffer, sizeof(titleBuffer), localizedFormat.c_str(), m_primaryFeatureName.c_str(), local_ref);
+        m_uiSubtitle = FormatSubtitle(true /* withType */);
+      }
+      else // no name but has local_ref
+      {
+        snprintf(titleBuffer, sizeof(titleBuffer), localizedFormat.c_str(), GetLocalizedType().c_str(), local_ref);
+        m_uiSubtitle = FormatSubtitle(false /* withType */);
+      }
+
+      m_uiTitle = titleBuffer;
+    }
+    else
+      SetStandardValues(out);
   }
   else if (IsBuilding())
   {
     bool const isAddressEmpty = m_address.empty();
     m_uiTitle = isAddressEmpty ? GetLocalizedType() : m_address;
     m_uiSubtitle = FormatSubtitle(!isAddressEmpty /* withType */);
   }
+  else
+    SetStandardValues(out);
+
+  m_hotelType = ftypes::IsHotelChecker::Instance().GetHotelType(ft);
+}
+
+void Info::SetStandardValues(feature::NameParamsOut out)
```

Comment:
nit: Please, keep inside the ```if``` like it was before. Better to read and seems like this ```if-else if-else if``` routing can be simplified. Additional function only hides the logic, IMO.

Reply:
it's in a function because it needs to be called from within the isPublicTransportStop statement, otherwise stops with no local_ref don't display like normal. 

i'll try and simplify it though

Reply:
idk, simplifying it makes it too abstract i think actually